Joy git cli vs gui (#897)
arcus/education_modules: git_cli_vs_gui/git_cli_vs_gui.md

No previous experience with git is required.
@end

@learning_objectives  
After completion of this module, learners will be able to:

- List an advantage and a disadvantage of using Git via a **graphical user interface**
- List an advantage and a disadvantage of using Git via a **command line interface**
- Name one major Git graphical user interface 

@end

good_first_module: false
data_task: data_management
collection: infrastructure_and_technology
coding_required: false
sequence_name: git_basics
previous_sequential_module: git_intro

@sets_you_up_for

- git_setup_mac_and_linux
- git_setup_windows

@end

@depends_on_knowledge_available_in

- git_intro
- demystifying_command_line

@end

@version_history 
Previous versions: 

No previous versions.

@end

import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
-->

# Git Command Line Interface versus Graphical User Interface

@overview

## What is Git?

You may already know what Git is and even use it for version control, but let's review.  We'll take on some of the text that Git uses to describe itself in its official website:

> [![The official git website, which states: "Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency. Git is easy to learn and has a tiny footprint with lightning fast performance. It outclasses SCM tools like Subversion, CVS, Perforce, and ClearCase with features like cheap local branching, convenient staging areas, and multiple workflows."](media/git_scm_com.png)](https://git-scm.com)

Author: @leemc-data-ed I ended up taking out "text" as well in this alt text... what say?

Reviewer: I like it!

<div class = "care">
<b style="color: rgb(var(--color-highlight));">A little encouragement...</b><br>

We just threw some Git jargon at you: **branch**, **commit**, **checkout**, **tracked**.  Don't worry if you don't know what these mean.  The important thing we want you to be aware of is what these commands look like when you're using Git in a CLI.

</div>

## Which is Better? 

Which is better when you use Git?  Graphical user interfaces, or a command line interface? It depends!  In the next section, we'll do some direct compare and contrast so you can get an idea of what the two approaches might look like in actual day to day work.  

Generally speaking, for everyday work in Git, you can choose whichever approach appeals to you more, so we'll present both the command line interface as well as a graphical user interface to allow you to see both. However, it's important to note that whichever interface you choose for daily work, you **might end up needing to use a command line interface for complex, out of the ordinary work** in Git.  Let's talk a bit about why this is the case.

> Git has lots of features, and designers can't include every possible command in an easy-to-use point and click GUI.  That means some things you may need to do in Git can **only** be done by using the command line, because they're so niche and specific to certain circumstances that graphical software simply hasn't included that task.  Still, there's a good chance that you might end up using a Git GUI for **years** before you run across a situation that requires the command line. 

Author: @leemc-data-ed  this is a bit of a combination of the two repetitive paragraphs.  What do you think?

Reviewer: This is great!

Command Line Interface
----

In a command line, you'd need to navigate to that directory (in the image below, in a MacOS Terminal, this is done using `cd`, which means "change directory"), and type `git init`.

![Terminal window which shows the command `cd my_important_directory` followed by the command `git init`.  The output of the second command reads "Initialized empty Git repository in /Users/paytonk/Documents/my_important_directory/.git/".](media/git_init.png)

Graphical User Interface
-----

In a GUI (here, GitHub Desktop), we have to put in the name of the repository, which is the name of the folder, as well as the name of the parent directory (`Documents`).

**Want to watch this process?  Click on the image below to play an animated gif.  It will continue to loop and you can re-start it by clicking again.**

<div style="display:none">

@gifPreload

</div>

<figure>

>   <img src="https://github.com/arcus/education_modules/blob/joy_git_cli_vs_gui/git_cli_vs_gui/media/git_init_gui.png?raw=true" height="756" width="540" alt="Adding a new repository using the GitHub Desktop graphical user interface" data-alt="https://github.com/arcus/education_modules/blob/joy_git_cli_vs_gui/git_cli_vs_gui/media/git_init_gui.gif?raw=true" style = "border: 1px solid rgb(var(--color-highlight));">

Reviewer: ```suggestion
  <img src="https://github.com/arcus/education_modules/blob/main/git_cli_vs_gui/media/git_init_gui.png?raw=true" height="756" width="540" alt="Adding a new repository using the GitHub Desktop graphical user interface" data-alt="https://github.com/arcus/education_modules/blob/main/git_cli_vs_gui/media/git_init_gui.gif?raw=true" style = "border: 1px solid rgb(var(--color-highlight));">
```

<!--
module_id: git_cli_vs_gui
author:   Joy Payton
email:    paytonk@chop.edu
version:  1.0.0
current_version_description: Initial version
module_type: standard
docs_version: 4.0.0
language: en
narrator: US English Female
mode: Textbook
title: Git Command Line Interface versus Graphical User Interface
comment:  Compare the two ways of interacting with Git to decide which is best for you.
long_description: There are two ways of using Git: (1) typing commands on a command line interface (CLI) or (2) clicking through options on a graphical user interface (GUI) software. This module will compare and contrast these two options to help you decide how to use Git in your work.
estimated_time_in_minutes: 30

@pre_reqs
Learners should know what version control is and why it's important and should also understand what a command line interface is and how that's different from a graphical user interface.  

To acquire basic knowledge of version control, we recommend our [Introduction to Version Control](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/git_intro/git_intro.md#1) module. 

> To acquire basic knowledge about command line interfaces, we recommend our [Demystifying the Command Line Interface](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/demystifying_command_line/demystifying_command_line.md) module.  

Reviewer: ```suggestion
To acquire basic knowledge about command line interfaces, we recommend our [Demystifying the Command Line Interface](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/demystifying_command_line/demystifying_command_line.md#1) module.  
```
I think it might be a good idea for us to start always adding the `#1` at the end of the module links-- then it always takes folks to the first page of a module, even if they have aggressive caching that might take them elsewhere (I've had this happen recently and been frustrated by it).

Author: Oh very good point!  I also find this stupidly annoying!

No previous experience with git is required.
@end

@learning_objectives  
After completion of this module, learners will be able to:

- List an advantage and a disadvantage of using Git via a **graphical user interface**
- List an advantage and a disadvantage of using Git via a **command line interface**
- Name one major Git graphical user interface 

@end

good_first_module: false
data_task: data_management
collection: infrastructure_and_technology
coding_required: false
sequence_name: git_basics
previous_sequential_module: git_intro

@sets_you_up_for

- git_setup_mac_and_linux
- git_setup_windows

@end

@depends_on_knowledge_available_in

- git_intro
- demystifying_command_line

@end

@version_history 
Previous versions: 

No previous versions.

@end

import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
-->

# Git Command Line Interface versus Graphical User Interface

@overview

## What is Git?

You may already know what Git is and even use it for version control, but let's review.  We'll take on some of the text that Git uses to describe itself in its official website:

> [![Screenshot of the official git website, which states: "Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency. Git is easy to learn and has a tiny footprint with lightning fast performance. It outclasses SCM tools like Subversion, CVS, Perforce, and ClearCase with features like cheap local branching, convenient staging areas, and multiple workflows."](media/git_scm_com.png)](https://git-scm.com)

Reviewer: ```suggestion
[![Text from the official git website, which states: "Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency. Git is easy to learn and has a tiny footprint with lightning fast performance. It outclasses SCM tools like Subversion, CVS, Perforce, and ClearCase with features like cheap local branching, convenient staging areas, and multiple workflows."](media/git_scm_com.png)](https://git-scm.com)
```
I'm basing this suggestion on our QA guidelines guidelines for alt text, which suggest that we not start alt text with "screenshot of"

Author: I literally committed the SAMPLE bad alt text ha!

Generally speaking, for everyday work in Git, you can choose whichever approach appeals to you more, so we'll present both the command line interface as well as a graphical user interface to allow you to see both. However, it's important to note that whichever interface you choose for daily work, you **might end up needing to use a command line interface for complex, out of the ordinary work** in Git.  Let's talk a bit about why this is the case.

While most things you want to accomplish on your computer can be done using multiple interfaces, some of which may be more user-friendly and graphical, there **are some things that can only be done by command line**.  Things that people need to do frequently, like look at a list of files in a directory, may be able to be accomplished through many interfaces, because designers built easy-to-use GUIs to help with these popular tasks.  Designers can't include every task in their GUI design, however, which means that very specific or customized tasks that are done more rarely may **only** be able to be accomplished by command line.  

> Git has lots of commands and actions, and some of them can only be done by using the command line, because they're so niche and specific to certain circumstances that graphical software simply hasn't included that task.  There's a good chance that you might end up using Git for **years** before you run across a situation that requires the command line, but it's important to know that it's possible you'll need to use the command line interface at some point.

Author: Oof, I agree.  Rewriting this.
